bitcoin/src/amount.rs

```diff
@@ -517,6 +517,16 @@ impl Amount {
     #[deprecated(since = "0.31.0", note = "Use Self::MIN instead")]
     pub const fn min_value() -> Amount { Amount(u64::min_value()) }
 
+    /// Creates an [`Amount`] from a value representing a whole number of bitcoin.
+    ///
+    /// This function handles an input arg that would overflow the inner type in the following ways:
+    ///
+    /// * In release mode non-const context: silently overflows
+    /// * In release mode const context: silently overflows
+    /// * In debug mode non-const context: panics
+    /// * In debug mode const context: fails to build
```

Comment:
Seems super-complex and hard to deal with, I'd prefer it to be always panic.

Reply:
Well, it's consistent with the behavior for every other integer overflow, so "super complex" or not, it's something users have to deal with.

But sure, I'd also be happy using array indexing to always panic. (I assume "always panic" will also translate into "won't compile" in const contexts?)

Reply:
> I assume "always panic" will also translate into "won't compile" in const contexts?

Yes.